cartographer/mapping/internal/3d/pose_graph_3d.h

```diff
@@ -199,6 +199,10 @@ class PoseGraph3D : public PoseGraph {
   void HandleWorkQueue(const constraints::ConstraintBuilder3D::Result& result)
       EXCLUDES(mutex_) EXCLUDES(work_queue_mutex_);
 
+  // Process pending tasks in the work queue on the calling thread, until the
+  // queue is empty.
+  void DrainWorkQueue() EXCLUDES(mutex_) EXCLUDES(work_queue_mutex_);
```

Comment:
DrainWorkQueue() does not exclude `mutex_` anymore, right?

Reply:
It has to exclude it because the work items will internally take `mutex_`.

Reply:
I did update the comment however since it was slightly inaccurate.